### PR TITLE
Adds support for running mono tests on Apple M1

### PR DESCRIPTION
**Pull request by steveisok**
Tweaked the mono build a bit to produce arm64 binaries & Enabled CI

### CODE REVIEW

File: eng/pipelines/libraries/helix-queues-setup.yml [hunk 1]
```diff
@@ -94,6 +94,10 @@ jobs:
       - ${{ if eq(parameters.jobParameters.isFullMatrix, false) }}:
         - OSX.1014.Amd64.Open
         - OSX.1015.Amd64.Open
+    
+    # OSX arm64
+    - ${{ if eq(parameters.platform, 'OSX_x64') }}:
```
Comment (directhex) [inline eng/pipelines/libraries/helix-queues-setup.yml:99]: This is a typo, the check should be for arm64 not x64

**Review by directhex — CHANGES_REQUESTED** [review]

File: eng/pipelines/libraries/helix-queues-setup.yml (hunk 1 above)
Comment (steveisok) [reply eng/pipelines/libraries/helix-queues-setup.yml:99]: Thanks.  I always do something like that ;-)

File: eng/pipelines/runtime.yml [hunk 2]
```diff
@@ -567,6 +568,7 @@ jobs:
     buildConfig: release
     platforms:
     - OSX_x64
+    - OSX_arm64
```
Comment (safern) [inline eng/pipelines/runtime.yml:571]: why do we need it in both, release and debug mono builds?

**Comment by sdmaclea** [comment]
I just triggered a rerun failed jobs since the M1 queues now seem to be functional.

File: eng/pipelines/libraries/helix-queues-setup.yml [hunk 3]
```diff
@@ -99,6 +99,10 @@ jobs:
       - ${{ if eq(parameters.jobParameters.isFullMatrix, false) }}:
         - OSX.1014.Amd64.Open
         - OSX.1015.Amd64.Open
+    
+    # OSX arm64
+    - ${{ if eq(parameters.platform, 'OSX_arm64') }}:
+      - OSX.1100.ARM64.Open
```
Comment (sdmaclea) [inline eng/pipelines/libraries/helix-queues-setup.yml:105]: This probably needs to be consolidated with lines 88-91 above.

**Pull request closed without merging**